source/agora/common/crypto/Key.d

```diff
@@ -348,12 +348,13 @@ public struct SecretKey
 
     public Signature sign (scope const(ubyte)[] msg) const nothrow @nogc
     {
-        Signature result;
-        // The second argument, `siglen_p`, a pointer to the length of the
-        // signature, is always set to `64U` and supports `null`
-        if (crypto_sign_ed25519_detached(result[].ptr, null, msg.ptr, msg.length, this.data[].ptr) != 0)
-            assert(0);
-        return result;
+        Pair kp = Pair.fromScalar(Scalar(this.data[]));
+        version (unittest)
+        {
+            const Pair noise = Pair.fromScalar(Scalar(hashMulti(kp.v, "sign")));
+            return agora.crypto.Schnorr.sign(kp, noise, msg);
+        }
```

Comment:
What is this ?

Reply:
This is for the unit test of `PrettyPrinter` checking the formatted output of a created block. A random noise for signing changes every time as you know. Is there any better way for that?

Reply:
I will take a look.

Reply:
Deleted with the support of @hewison-chris.